File: JAVA Programming/SelectionSort.java
public class SelectionSort {
    public static void Selection_Sort(int A[], int n) {
        int i, j, min;

        for (i = 0; i < n - 1; i++) {
            min = i;
            for (j = i + 1; j < n; j++) {
                if (A[j] < A[min]) {
                    min = j;
                }
            }

            int temp = A[min];
            A[min] = A[i];
            A[i] = temp;
        }
    }

    public static void main(String[] args) {

        int arr[] = { 4, 7, 9, 21, 3, 4, 5, 1 }; // size 8 ;
        int n = 8;
        System.out.println("Print Array Before Sorting ");
        for (int e : arr) {

            System.out.print("\t" + e);

        }

        System.out.println("\nPrint Array After Sorting ");
        Selection_Sort(arr, n);
        for (int e : arr) {

            System.out.print("\t" + e);

        }
    }

}

File: C++ Programming/Selection_sort.cpp
#include <iostream>
using namespace std;
void Print(int *arr, int size)
{
    int i = 0;
    cout << "The Array is --- \n";
    for (i = 0; i < size; i++)
        cout << arr[i] << "\t";
    cout << endl
         << endl;
}
void SelectionSort(int *arr, int size)
{
    int i, j, min, temp;
    for (i = 0; i < size - 1; i++)
    {
        min = i;
        for (j = i + 1; j < size; j++)
            if (arr[min] > arr[j])
                min = j;
        if (arr[min] != arr[i])
        {
            temp = arr[min];
            arr[min] = arr[i];
            arr[i] = temp;
        }
    }
}
int main(int argc, char const *argv[])
{
    int size = 10;
    int arr[size] = {1,56,23,12,5,67,321,567,53,20,37};
    Print(arr,size);
    cout<<"After the Selection sort \n";
    SelectionSort(arr,size);
    Print(arr,size);
    return 0;
}
